src/components/Stakeholders/Filter.js

```diff
@@ -0,0 +1,342 @@
+  const [open, setOpen] = useState(false);
+  const [filters, setFilters] = useState({});
+  const orgNames = new Set(['']);
+  const firstNames = new Set(['']);
+  const lastNames = new Set(['']);
+  const emails = new Set(['']);
+  const phones = new Set(['']);
+  const websites = new Set(['']);
+  const maps = new Set(['']);
+
+  // create filter lists
+  useEffect(() => {
+    stakeholders?.forEach((org) => {
+      org.org_name && orgNames.add(org.org_name);
+      org.first_name && firstNames.add(org.first_name);
+      org.last_name && lastNames.add(org.last_name);
+      org.email && emails.add(org.email);
+      org.phone && phones.add(org.phone);
+      org.website && websites.add(org.website);
+      org.map && maps.add(org.map);
+    });
+    setFilters({
+      orgNames: Array.from(orgNames).sort(),
+      firstNames: Array.from(firstNames).sort(),
+      lastNames: Array.from(lastNames).sort(),
+      emails: Array.from(emails).sort(),
+      phones: Array.from(phones).sort(),
+      websites: Array.from(websites).sort(),
+      maps: Array.from(maps).sort(),
+    });
+  }, [stakeholders?.length]);
+
+  const close = () => {
+    setFormData(filter);
+    setOpen(false);
+  };
+
+  const handleChanges = (e) => {
+    const key = e.target.name === 'organization_id' ? 'id' : e.target.name;
+    const value =
+      e.target.name === 'organization_id'
+        ? orgList.find((o) => o.stakeholder_uuid === e.target.value)
+            ?.stakeholder_uuid || ''
+        : e.target.value !== 'Not Set'
+        ? e.target.value
+        : '';
+    setFormData({ ...formData, [key]: value });
+  };
+
+  const handleEnterPress = (e) => {
+    e.key === 'Enter' && applyFilters(e);
+  };
+
+  const resetFilters = () => {
+    setFormData(initialFilterState);
+    const resetFilter = new FilterModel(initialFilterState);
+    updateFilter(resetFilter);
+  };
+
+  const applyFilters = () => {
+    const newFilter = new FilterModel({
+      ...filter,
+      ...formData,
+    });
+    updateFilter(newFilter);
+    setOpen(false);
+  };
+
+  const defaultOrgList = [
+    {
+      id: ORGANIZATION_NOT_SET,
+      name: 'Not Set',
+      value: '',
+    },
+  ];
+
+  const defaultTypeList = [
+    {
+      name: 'Not set',
+      value: undefined,
+    },
+    {
+      name: 'Organization',
+      value: 'Organization',
+    },
+    {
+      name: 'Person',
+      value: 'Person',
+    },
+  ];
+
+  return (
+    <>
+      <TextField
+        id="search"
+        name="search"
+        label="Search"
+        variant="outlined"
+        value={formData.search}
+        onChange={handleChanges}
+        autoComplete="off"
+        onKeyDown={handleEnterPress}
+      />
+      <Button
+        variant="text"
+        color="primary"
+        onClick={() => setOpen(true)}
+        startIcon={<FilterIcon />}
+        className={classes.ml}
+      >
+        Filter
+      </Button>
+      {Object.keys(filter).length > 0 && (
```

Comment:
`filter` always has a complete set of keys with empty string values by default, so this always shows a button.